[shopsys] added blackfire profiling configuration and instructions (#3168)
shopsys/shopsys: docs/introduction/blackfire-profiling.md

# Blackfire profiling

[TOC]

## Introduction

Blackfire is a powerful tool for profiling and debugging that helps you to understand the performance of your application.
In the Shopsys Platform, Blackfire is preinstalled for local development, allowing developers to easily profile and optimize their code.

## Prerequisites

Before you can start using Blackfire in the Shopsys Platform, you need to have the following:

-   Blackfire account
    -   You can create an account on the [Blackfire website](https://blackfire.io/).

> -   Blackfire browser extension installed (optional)
>     -   The extension is available for [Google Chrome](https://docs.blackfire.io/integrations/browsers/chrome) and [Mozilla Firefox](https://docs.blackfire.io/integrations/browsers/firefox).

Reviewer: ```suggestion
-   [Blackfire browser extension](https://docs.blackfire.io/integrations/browsers/index) installed (optional)
```

I think this way it is better, so we do not have to update our docs if they introduce extension for other browsers.

## Configuration

Edit the `docker-compose.yml` file in your Shopsys Platform project to include the Blackfire credentials.
Uncomment the environment variables for Blackfire and fill in the values with your actual Blackfire credentials.

```yaml
php-fpm:
    build:
        context: .
        dockerfile: project-base/app/docker/php-fpm/Dockerfile
        target: development
        args:
            www_data_uid: 1000
            www_data_gid: 1000
            project_root: project-base/app
    container_name: shopsys-framework-php-fpm
    volumes:
        - ~/.gitconfig:/home/www-data/.gitconfig
        - .:/var/www/html
    # Uncomment the following lines and fill values to enable Blackfire
    environment:
        BLACKFIRE_SERVER_ID: your_server_id
        BLACKFIRE_SERVER_TOKEN: your_server_token
        BLACKFIRE_CLIENT_ID: your_client_id
        BLACKFIRE_CLIENT_TOKEN: your_client_token
```

Replace `your_client_id`, `your_client_token`, `your_server_id`, and `your_server_token` with your actual Blackfire credentials.

After updating the `docker-compose.yml` file, restart your Docker containers to apply the changes.

```bash

> docker-compose up -d --force-recreate php-fpm

Reviewer: ```suggestion
docker compose up -d --force-recreate php-fpm
```

```

## Profiling CLI Scripts

You can profile your CLI scripts using the Blackfire CLI tool preinstalled in the `php-fpm` container.
For example, to profile a CLI command, use the following command:

```bash

> docker-compose exec -it php-fpm blackfire run php bin/console your:command

Reviewer: ```suggestion
docker compose exec -it php-fpm blackfire run php bin/console your:command
```

```

This will run the specified CLI command and profile its execution.
You can view the profiling results in your Blackfire account.

## Profiling HTTP Requests

> To profile HTTP requests, you need to install the Blackfire browser extension.
> The extension is available for [Google Chrome](https://docs.blackfire.io/integrations/browsers/chrome) and [Mozilla Firefox](https://docs.blackfire.io/integrations/browsers/firefox).

Reviewer: ```suggestion
To profile HTTP requests, you need to install the [Blackfire browser extension](https://docs.blackfire.io/integrations/browsers/index).
```

I think this way it is better, so we do not have to update our docs if they introduce extension for other browsers.
